Source/Libraries/FaultData/DataWriters/Emails/EventEmailService.cs

```diff
@@ -378,16 +377,21 @@ private void LoadSentEmail(EmailType email, DateTime now, List<string> recipient
             }
         }
 
-        public void ApplyChartTransform(List<Attachment> attachments, XDocument htmlDocument)
+        public void ApplyChartTransform(List<Attachment> attachments, XDocument htmlDocument, int minSamplesPerCycle = -1)
         {
             using (AdoDataConnection connection = ConnectionFactory())
             {
                 htmlDocument.TransformAll("chart", (element, index) =>
                 {
-                    string chartEventID = (string)element.Attribute("eventID") ?? "-1";
+                    string chartEventID = (string) element.Attribute("eventID") ?? "-1";
                     string cid = $"event{chartEventID}_chart{index:00}.png";
 
-                    Stream image = ChartGenerator.ConvertToChartImageStream(connection, element);
+                    string stringMinimum = (string) element.Attribute("minimumSamplesPerCycleOverride");
+                    int passedMinimum = minSamplesPerCycle;
+                    if (!(stringMinimum is null))
+                        int.TryParse(stringMinimum, out passedMinimum); //passedMinimum will be 0 if this fails
```

Comment:
This is kind of a nitpicky code quality thing, but I recommend expressing your intent directly through code rather than calling out edge cases in comments. This way we can apply the default value of -1 (or possibly throw an exception) instead of using the value incidentally assigned by `int.TryParse()`, which makes more sense semantically even though the two values effectively do the same thing.

```suggestion
                    if (!(stringMinimum is null) && !int.TryParse(stringMinimum, out passedMinimum))
                        passedMinimum = -1;
```